### PR TITLE
Implement ItemsHandling flags.

### PR DESCRIPTION
**Pull request by Ijwu**
Implements https://github.com/ArchipelagoMW/Archipelago/pull/227

### CODE REVIEW

File: Archipelago.MultiClient.Net/ArchipelagoSession.cs [hunk 1]
```diff
@@ -80,7 +81,7 @@ private void Socket_PacketReceived(ArchipelagoPacketBase packet)
         ///     The connect attempt is synchronous and will lock for up to 5 seconds as it attempts to connect to the server. 
         ///     Most connections are instantaneous however the timeout is 5 seconds before it returns <see cref="T:Archipelago.MultiClient.Net.LoginFailure"/>.
         /// </remarks>
-        public LoginResult TryConnectAndLogin(string game, string name, Version version, List<string> tags = null, string uuid = null, string password = null)
+        public LoginResult TryConnectAndLogin(string game, string name, Version version, List<string> tags = null, string uuid = null, string password = null, ItemsHandlingFlags itemsHandlingFlags = ItemsHandlingFlags.RemoteItems)
```
Comment (JarnoWesthof) [inline Archipelago.MultiClient.Net/ArchipelagoSession.cs:84]: I would recommend to move this to before tags, and make it madatory decision for a client to choose what to use when they upgrade, else clients might upgrade and if they forget to choose an ItemsHandlingFlags thier builds will still pass without them knowing they just opted in for 0b001

**Review by JarnoWesthof — CHANGES_REQUESTED** [review]

File: Archipelago.MultiClient.Net/ArchipelagoSession.cs (hunk 1 above)
Comment (Ijwu) [reply Archipelago.MultiClient.Net/ArchipelagoSession.cs:84]: I do want to enforce the update, however I wanted the method signature to remain backwards compatible. This was the compromise I came up with.

Making it nullable might work as the most backwards compatible but null value is deprecated in AP so we'd have to remove it eventually anyway.

Comment (JarnoWesthof) [reply Archipelago.MultiClient.Net/ArchipelagoSession.cs:84]: why do you want the method call to be backward compatiable? it seems logical to me if make it fail on build so implementers know they have to supply the parameter

Comment (Ijwu) [reply Archipelago.MultiClient.Net/ArchipelagoSession.cs]: I've made an overload which requires the flags to be set. Then I obsoleted the old. I made the flags nullable and made the old overload default to null.

Nullable enums will be removed when we remove the obsolete version.

Comment (JarnoWesthof) [reply Archipelago.MultiClient.Net/ArchipelagoSession.cs]: I made the ItemsHandlingFlags not nullable in the new method signature

**Review by JarnoWesthof — APPROVED** [review]

**Pull request merged**